core/common/src/main/java/alluxio/worker/block/io/BlockWriter.java

```diff
@@ -37,4 +39,14 @@
    * @return channel
    */
   GatheringByteChannel getChannel();
+
+  /**
+   * Transfers buf.writableBytes() bytes to the this block writer from the given buf.
+   *
+   * @param buf the byte buffer to hold the data
+   * @return the number of bytes transferred, -1 if the end of file has reached before reading
+   *         any data
+   * @throws IOException if any I/O errors occur
+   */
+  int transferFrom(ByteBuf buf) throws IOException;
```

Comment:
From the usage of this method, it doesn't seem like there is a reason for it to return an int